tests/extmod/vfs_posix_ilistdir_filter.py

```diff
@@ -0,0 +1,45 @@
+# Test ilistdir filter of . and .. for VfsPosix.
+
+try:
+    import os
+
+    os.VfsPosix
+except (ImportError, AttributeError):
+    print("SKIP")
+    raise SystemExit
+
+
+def test(testdir):
+    vfs = os.VfsPosix(testdir)
+
+    expected = {".a": True, "..a": True, "...a": True, "a.": True}
+    actual = {}
+
+    for dir in expected:
+        vfs.mkdir(dir)
+
+    for entry in vfs.ilistdir("/"):
+        actual[entry[0]] = True
+
+    assert expected == actual
```

Comment:
In MicroPython tests, we usually `print()` and compare to the .exp file instead of `assert` since MicroPython doesn't provide useful information as to what caused the failure with `assert`.

Reply:
Well, I was worried about the order the files would be returned by ilistdir().  That is why I chose to put the data into a dictionary so that the test would pass regardless of what the order was returned by ilistdir() on a particular platform.

I guess I could push onto an array and print a sorted array of file names and put the sorted list in the .exp file.  I do think we would need something to guarantee order or we may get failures due to platform differences in how directory entries happen to be returned.